torchvision/io/_video_opt.py

```diff
@@ -416,7 +429,9 @@ def _read_video_from_memory(
     return vframes, aframes
 
 
-def _read_video_timestamps_from_memory(video_data):
+def _read_video_timestamps_from_memory(
+    video_data: Union[torch.Tensor, BufferedIOBase]
```

Comment:
we should not forget that those type signatures are also used for torchscript, and torchscript doesn't support Union, and I'm not sure about `BufferedIOBase`, so let's just leave `Tensor` here.